### ISSUE TITLE
Error - UIAlertController:supportedInterfaceOrientations was invoked recursively!

### ISSUE THREAD

**Issue by kapso** [GantMan/RedAlert#20]
Getting this error

015-09-18 00:53:41.970 App[27430:7325717] *** Assertion failure in -[UIAlertController supportedInterfaceOrientations], /BuildRoot/Library/Caches/com.apple.xbs/Sources/UIKit_Sim/UIKit-3505.16/UIAlertController.m:541
2015-09-18 00:53:42.265 App[27430:7325717] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'UIAlertController:supportedInterfaceOrientations was invoked recursively!'

When I run this code:
```ruby
rmq.app.alert(title: "", message: "Enter valid billing ZIP code", style: :input) do |action_type, fields|
  @cc_zip_code = fields[:text].text
end
```

Gem version & Ruby Motion 4.0 / Xcode 7 / iOS 9
```ruby
gem "RedAlert", "1.3"
gem "ruby_motion_query", "1.6.1"
```

**Comment by GantMan** [comment]
I was on an older version of all 3 to deal with a legacy client.   I'll upgrade one of my machines and check this out.  Thanks for the report.

**Comment by kapso** [comment]
@GantMan 

Thanks. I just downgraded to 3.14, but still get the same error. However noticed the issues is because of this gem https://github.com/infinitered/ProMotion

gem "ProMotion", "1.2.1" # This works
gem "ProMotion", "2.4.2" # This does not

**Comment by eahanson** [comment]
I'm having the same issue. 

Xcode Version 6.4 (6E35b), ProMotion 2.4.2, RedAlert 1.3. Works when running on iOS 8, fails when running on iOS 9.

**Comment by GantMan** [comment]
I've got time scheduled with Jamon to nail this down tomorrow.

**Comment by andrewhavens** [comment]
[Based on these answers on StackOverflow](http://stackoverflow.com/questions/31406820/uialertcontrollersupportedinterfaceorientations-was-invoked-recursively), it looks like there's a fix by monkey-patching UIAlertController. However, I couldn't figure out how to do it in RubyMotion.

**Comment by GantMan** [comment]
great find @andrewhavens !

**Comment by bmichotte** [comment]
When I had this issue, I found the link @andrewhavens refers.

This is the solution I used... and it's working

```ruby
# ui_alert_controller.rb
class UIAlertController
	
	def supportedInterfaceOrientations
		UIInterfaceOrientationMaskLandscape
	end

	def shouldAutorotate
		true
	end
end
```

There are a few caveats
1) if you use something like `if Kernel.const_defined?("UIAlertController")` to check if you are on iOS 8+, you have to add `&& UIAlertController.respond_to?(:'alertControllerWithTitle:message:preferredStyle:')` because now, `UIAlertController` exists
2) my app only supports landscape, I guess you have to change the `supportedInterfaceOrientations` method for something like
```ruby
def supportedInterfaceOrientations
   rmq.device.landscape? ? UIInterfaceOrientationMaskLandscape : UIInterfaceOrientationMaskPortrait
end
```

**Comment by GantMan** [comment]
Hey guys, can you please try out this branch?

https://github.com/GantMan/RedAlert/tree/ios9_fix

I found crashes in internationalization in some cases, and also found some strange other issues which I hopefully fixed.

**Comment by andrewhavens** [comment]
@GantMan I tried your branch and it works. Also, my question about whether or not "portrait-only" would be respected is true. It seems that if you set `app.interface_orientations = [:portrait]` in your Rakefile, then `rmq.device.landscape?` is never true because the app never allows itself to be rotated. So all is good. :+1:

**Comment by derekgreenberg** [comment]
I've verified this fix in RedAlert 1.4 on an iOS 9 device. It's included in redpotion 1.4.0.
